righttyper/righttyper_types.py

```diff
@@ -230,21 +230,22 @@ def process(self) -> CallTrace:
             
         type_data = (*self.args, retval)
 
-        if self.self_type:
+        if self.self_type and self.self_replacement:
+            self_type = cast(TypeInfo, self.self_type)
+            self_replacement = cast(TypeInfo, self.self_replacement)
+
             class SelfTransformer(TypeInfo.Transformer):
                 """Replaces 'self' types with the type of the class that defines them,
                    also setting is_self for possible later replacement with typing.Self."""
 
                 def visit(vself, node: TypeInfo) -> TypeInfo:
-#                    if self.self_type: print(f"checking {str(node)} against {str(self.self_type)}")
+#                    if self_type: print(f"checking {str(node)} against {str(self_type)}")
```

Comment:
Remove cruft?

Reply:
I guess... I left it in because I found myself rewriting it too many times.